[Bug]: macOS: PyPy 3.8 (v7.3.9) threading get_native_id Broken

### Bug summary

In PyPy 3.8 (v7.3.9), matplotlib 3.6.0 is cannot be imported, leading to breakage.

Ref.: https://foss.heptapod.net/pypy/pypy/-/issues/3764

### Code for reproduction

```python
import matplotlib.pyplot as plt
```


### Actual outcome

```
AttributeError: module 'threading' has no attribute 'get_native_id'
```
in `matplotlib/pyplot.py:337`
```
     def _warn_if_gui_out_of_main_thread():
         # This compares native thread ids because even if python-level Thread
         # objects match, the underlying OS thread (which is what really matters)
         # may be different on Python implementations with green threads.
         if (_get_required_interactive_framework(_get_backend_mod()) and
 >               threading.get_native_id() != threading.main_thread().native_id):
 E               AttributeError: module 'threading' has no attribute 'get_native_id'
```

### Expected outcome

Work as usual.

### Additional information

This has been first posted in PyPy: https://foss.heptapod.net/pypy/pypy/-/issues/3764

I saw this on Conda-Forge, starting a few weeks ago as they shipped the latest PyPy 3.8.

### Operating system

macOS 11.7 (x86_64) via Azure Pipelines 14 Runner

### Matplotlib Version

3.6.0

### Matplotlib Backend

default

### Python version

PyPy 3.8 (v7.3.9)

### Jupyter version

N/A

### Installation

conda

--- a/lib/matplotlib/pyplot.py
+++ b/lib/matplotlib/pyplot.py
@@ -332,11 +332,21 @@ def switch_backend(newbackend):
 
 
 def _warn_if_gui_out_of_main_thread():
-    # This compares native thread ids because even if python-level Thread
-    # objects match, the underlying OS thread (which is what really matters)
-    # may be different on Python implementations with green threads.
-    if (_get_required_interactive_framework(_get_backend_mod()) and
-            threading.get_native_id() != threading.main_thread().native_id):
+    warn = False
+    if _get_required_interactive_framework(_get_backend_mod()):
+        if hasattr(threading, 'get_native_id'):
+            # This compares native thread ids because even if Python-level
+            # Thread objects match, the underlying OS thread (which is what
+            # really matters) may be different on Python implementations with
+            # green threads.
+            if threading.get_native_id() != threading.main_thread().native_id:
+                warn = True
+        else:
+            # Fall back to Python-level Thread if native IDs are unavailable,
+            # mainly for PyPy.
+            if threading.current_thread() is not threading.main_thread():
+                warn = True
+    if warn:
         _api.warn_external(
             "Starting a Matplotlib GUI outside of the main thread will likely "
             "fail.")

